corehq/apps/linked_domain/updates.py

```diff
@@ -429,6 +433,12 @@ def update_user_roles(domain_link, is_pull=False, overwrite=False):
             continue
 
         permissions = HqPermissions.wrap(upstream_role_def["permissions"])
+        if permissions.web_apps_list:
+            permissions.web_apps_list = [
+                downstream_id for downstream_id, upstream_id in app_id_map.items()
+                if upstream_id in permissions.web_apps_list
+            ]
```

Comment:
I assume this new usage of app ids does not require any data migrations since this is a new feature that is only deployed on staging where old data either does not exist or does not matter. Is that right?

Reply:
Yes, exactly.